stock_picking_production_lot_manual_selector/tests/test_stock_picking_lot.py

```diff
@@ -0,0 +1,78 @@
+
+from odoo.exceptions import UserError
+from odoo.tests import Form
+
+from odoo.addons.stock.tests.test_move import StockMove
+
+
+class TestStockLockLot(StockMove):
+    def test_manual_lot(self):
+        self.env["stock.quant"]._update_available_quantity(
+            self.product_serial, self.stock_location, 1
+        )
+        serial1 = self.env["stock.production.lot"].create(
+            {
+                "name": "Serial1",
+                "product_id": self.product_serial.id,
+                "company_id": self.env.company.id,
+            }
+        )
+        serial2 = self.env["stock.production.lot"].create(
+            {
+                "name": "Serial2",
+                "product_id": self.product_serial.id,
+                "company_id": self.env.company.id,
+            }
+        )
+        picking_type = self.env["stock.picking.type"].create(
+            {
+                "name": "Manual Lot Selection",
+                "code": "incoming",
+                "sequence_code": "IN",
+                "use_create_lots": False,
+                "use_existing_lots": False,
+                "use_manual_lots": True,
+                "default_location_src_id": self.stock_location.id,
+                "default_location_dest_id": self.customer_location.id,
+            }
+        )
+        customer = self.env["res.partner"].create({"name": "SuperPartner"})
+        picking = self.env["stock.picking"].create(
+            {
+                "location_id": self.stock_location.id,
+                "location_dest_id": self.customer_location.id,
+                "partner_id": customer.id,
+                "picking_type_id": picking_type.id,
+            }
+        )
+        move = self.env["stock.move"].create(
+            {
+                "name": "SuperMove01",
+                "location_id": picking.location_id.id,
+                "location_dest_id": picking.location_dest_id.id,
+                "picking_id": picking.id,
+                "product_id": self.product_serial.id,
+                "product_uom_qty": 1,
+                "product_uom": self.product_serial.uom_id.id,
+            }
+        )
+        picking.action_confirm()
+        with Form(picking) as picking_form:
+            with picking_form.move_line_ids_without_package.new() as line_form:
+                line_form.product_id = move.product_id
+                line_form.qty_done = 1
+                line_form.lot_id = serial1
+            picking_form.save()
+
+            with self.assertRaises(UserError):
+                picking._action_done()
+
+            with picking_form.move_line_ids_without_package.edit(0) as line_form:
+                line_form.manual_production_lot_id = serial2
+                self.assertEqual(line_form.lot_id, serial2)
+                self.assertEqual(line_form.manual_production_lot_id, serial2)
+                picking_form.save()
+
+            picking._action_done()
```

Comment:
please add another test with the sub locations to make sure that the stock does not get negative (see  [this comment](https://github.com/OCA/stock-logistics-workflow/pull/1661/files#diff-32b1a61db5271cbeec234d6f1cf2c099639a79c8028d289a0fa2f2510e4d22ebR27))